[Failing Test]: Python PostCommit xlang jdbc tests failing due to breaking change introduced in #24851

### What happened?

```
      with ExternalTransform.service(self._expansion_service) as service:
        response = service.Expand(request)
        if response.error:
>         raise RuntimeError(response.error)
E         RuntimeError: java.lang.IllegalArgumentException: Cannot find field maxConnections in schema Fields:
E         Field{name=driverClassName, description=, type=STRING NOT NULL, options={{}}}
E         Field{name=jdbcUrl, description=, type=STRING NOT NULL, options={{}}}
E         Field{name=username, description=, type=STRING NOT NULL, options={{}}}
E         Field{name=password, description=, type=STRING NOT NULL, options={{}}}
E         Field{name=connectionProperties, description=, type=STRING, options={{}}}
E         Field{name=connectionInitSqls, description=, type=ITERABLE, options={{}}}
E         Field{name=readQuery, description=, type=STRING, options={{}}}
E         Field{name=writeStatement, description=, type=STRING, options={{}}}
E         Field{name=fetchSize, description=, type=INT16, options={{}}}
E         Field{name=outputParallelization, description=, type=BOOLEAN, options={{}}}
E         Field{name=autosharding, description=, type=BOOLEAN, options={{}}}
E         Encoding positions:
E         {connectionProperties=4, autosharding=10, password=3, writeStatement=7, outputParallelization=9, fetchSize=8, driverClassName=0, jdbcUrl=1, connectionInitSqls=5, username=2, readQuery=6}
E         Options:{{}}UUID: 49358cda-6f26-4080-86fc-36fcc5d52b44
E         	at org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument(Preconditions.java:440)
E         	at org.apache.beam.sdk.schemas.Schema.indexOf(Schema.java:1365)
E         	at org.apache.beam.sdk.values.Row.getInt32(Row.java:154)
E         	at org.apache.beam.sdk.io.jdbc.JdbcSchemaIOProvider$JdbcSchemaIO.getDataSourceConfiguration(JdbcSchemaIOProvider.java:178)
E         	at org.apache.beam.sdk.io.jdbc.JdbcSchemaIOProvider$JdbcSchemaIO$2.expand(JdbcSchemaIOProvider.java:144)
E         	at org.apache.beam.sdk.io.jdbc.JdbcSchemaIOProvider$JdbcSchemaIO$2.expand(JdbcSchemaIOProvider.java:139)
```

### Issue Failure

Failure: Test is continually failing

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [X] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java
+++ b/sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcSchemaIOProvider.java
@@ -175,9 +175,11 @@ public class JdbcSchemaIOProvider implements SchemaIOProvider {
         dataSourceConfiguration = dataSourceConfiguration.withConnectionInitSqls(initSqls);
       }
 
-      @Nullable Integer maxConnections = config.getInt32("maxConnections");
-      if (maxConnections != null) {
-        dataSourceConfiguration = dataSourceConfiguration.withMaxConnections(maxConnections);
+      if (config.getSchema().hasField("maxConnections")) {
+        @Nullable Integer maxConnections = config.getInt32("maxConnections");
+        if (maxConnections != null) {
+          dataSourceConfiguration = dataSourceConfiguration.withMaxConnections(maxConnections);
+        }
       }
 
       return dataSourceConfiguration;

